src/test/java/catan/controllers/user/UserControllerTest.java

```diff
@@ -141,4 +155,53 @@ public void shouldLogoutSuccessfullyWithHttpCode200WhenAnyTokenIsPassed() {
                 statusCode(200).
                 body(isEmptyString());
     }
+
+    // end of code which needs to be refactored (see the first line of the file)
+
+
+
+    @Test
+    public void should_successfully_register_and_login_guest() {
+        String userToken = registerAndLoginGuest(USER_NAME_GUEST_1)
+                .then()
+                .statusCode(200)
+                .body("token", not(equalTo(isEmptyString())))
+                .extract()
+                .path("token");
+
+        getUserDetails(userToken)
+                .then()
+                .statusCode(200)
+                .body("guest", equalTo(true));
+
+    }
+
+    @Test
+    public void should_fail_when_register_and_login_guest_if_username_is_taken_by_other_guest() {
+        registerAndLoginGuest(USER_NAME_GUEST_2);
+
+        registerAndLoginGuest(USER_NAME_GUEST_2)
+                .then()
+                .statusCode(400)
+                .body("errorCode", equalTo("USERNAME_ALREADY_EXISTS"));
+    }
+
+    @Test
+    public void should_fail_when_register_and_login_guest_if_username_is_taken_by_regular_user() {
+        registerUser(USER_NAME_3, USER_PASSWORD_3);
+
+        registerAndLoginGuest(USER_NAME_3)
+                .then()
+                .statusCode(400)
+                .body("errorCode", equalTo("USERNAME_ALREADY_EXISTS"));
+    }
+
+    @Test
+    public void should_fail_when_register_and_login_guest_if_username_is_empty() {
+        registerAndLoginGuest("")
+                .then()
+                .statusCode(400)
+                .body("errorCode", equalTo("ERROR"));
+    }
+
```

Comment:
how can he login as a regular user? first of all we don;t know it as there is no such test case.
the seccond, if we allow user login for guests as general user, what password should he input? if it is empty, anyone will be able to login with his username))